Competition/src/main/java/competition/RumbleManagerCommand.java

```diff
@@ -25,19 +28,26 @@ public RumbleManagerCommand(VisionSubsystem visionSubsystem, RumbleManager rumbl
         propFactory.setPrefix(this.getPrefix());
         rumbleIntensity = propFactory.createPersistentProperty("RumbleIntensity", .5);
         rumbleLength = propFactory.createPersistentProperty("RumbleLength", .1);
+        timeBetweenRumble = propFactory.createPersistentProperty("TimeBetweenRumbles", 60);
     }
 
     @Override
     public void initialize() {
         log.info("Initializing");
         wasTargetInView = false;
+        lastSawTarget = XTimer.getFPGATimestamp();
     }
 
     @Override
     public void execute() {
         boolean isTargetInView = visionSubsystem.isTargetInView();
         if (visionSubsystem.isTargetInView() && wasTargetInView == false) {
             rumble.rumbleDriverGamepad(rumbleIntensity.get(), rumbleLength.get());
+            if (XTimer.getFPGATimestamp() - lastSawTarget < timeBetweenRumble.get()) {
```

Comment:
I don't think this logic works. This is an area where you may need some tests in order to get everything working properly.

Reply:
It might help to draw out all the states on a whiteboard - there's a lot going on here.
* Calculating if we are in a rising edge
* Calculating how long it has been since the last successful rumble call
* Only calling the rumble subsystem on the rising edge of the target in view
* Additionally, not calling the rumble subsystem if we called the rumble subsystem recently